R/update-news.R

```diff
@@ -354,6 +355,35 @@ default_type <- function() {
   "Uncategorized"
 }
 
+capitalize_description <- function(df) {
+  df$description
+
+  # leave code alone
+  if (substr(df$description, 1, 1) == "`") {
+    return(df)
+  }
+
+  # leave package name alone
+  pkg_name <- desc::desc_get("Package")
+  start_with_pkg <- (grepl(sprintf("^%s ", pkg_name), df$description) ||
+    grepl(sprintf("^%s's", pkg_name), df$description))
+  if (start_with_pkg) {
+    return(df)
+  }
+
+  df$description <- paste0(
+    toupper(substr(df$description, 1, 1)),
+    substr(df$description, 2, nchar(df$description))
+  )
+  df
+}
+
+capitalize_news <- function(news_items) {
+  news_item <- split(news_items, seq_len(nrow(news_items))) %>%
+    purrr::map(capitalize_description)
+  do.call(rbind, news_item)
```

Comment:
I don't understand the need to split here.

`map_dfr()` ?

Reply:
`map_dfr()` requires dplyr which fledge does not depend upon hence the awkward split.
Should I define a helper `map_dfr` that would hide the split?

Reply:
(I added such a helper)